### DISCUSSION TITLE
How to ignore Makefiles?

### DISCUSSION THREAD

**Discussion by Therkelsen** [github-linguist/linguist#5356] [Q&A] [answered]
Hey there guys

I have a repo for my semester project at school, and it says it's majorily a Make repo due to the damn Makefiles that are generated.
Can I somehow not vendor all Makefiles? I've tried in the .gitattributes like this:
```
# Auto detect text files and perform LF normalization
* text=auto
*.stash linguist-vendored=true
*.debug linguist-vendored=true
*.release linguist-vendored=true
*/Makefile linguist-vendored=true
```

I honestly though the */Makefile would do the trick, is there another way I could do it?

**Comment by lildude** [comment, marked as answer]
> `*/Makefile linguist-vendored=true`
> I honestly though the */Makefile would do the trick, is there another way I could do it?

You're almost there with this one. It's not quite doing what you want because `*` doesn't recurse (see previous [discussions](https://github.com/github/linguist/discussions?discussions_q=recursive+OR+recurse)) so it will only match files like `foo/Makefile`, `bar/Makefile`, but not `ano/dir/Makefile`. I suspect all your files fall into the latter category.

To get the recursion you'd need to either use: `**/Makefile linguist-vendored=true` or just `Makefile linguist-vendored=true`.

As an aside, the `=true` is also optional so `Makefile linguist-vendored` will do the trick too.

**Reply by Therkelsen** [reply]
Perfect! Thank you so much :)
